mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -179,7 +230,7 @@ abstract class ClassificationModel[FeaturesType, M <: ClassificationModel[Featur
         }
         predictUDF(col(getFeaturesCol))
       }
-      outputData = outputData.withColumn(getPredictionCol, predUDF)
+      outputData = outputData.withColumn(getPredictionCol, predUDF, predictionMetadata)
```

Comment:
The problem is that the name may be different.  transform can be given a new output column name, so we may need to update the name